modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -168,8 +168,10 @@ object GlobalL0Service {
           } { lastStoredOrdinal =>
             for {
               l0Peer <- globalL0ClusterStorage.getRandomPeer
-              lastOrdinal <- l0GlobalSnapshotClient.getLatestOrdinal.run(l0Peer)
-              pulled <- pullSnapshots(l0Peer, lastStoredOrdinal.next, lastOrdinal)
+              latestOrdinal <- l0GlobalSnapshotClient.getLatestOrdinal.run(l0Peer)
+              nextOrdinal = lastStoredOrdinal.next
+              lastOrdinal = calculateLastOrdinal(nextOrdinal, latestOrdinal)
```

Comment:
I would find a better name for these, latest refers to the latest on the node we query, last is the last one we want to download (limit?). The names themselves don't suggest which is which.